vpr/src/place/place.cpp

```diff
@@ -1491,14 +1717,43 @@ static e_move_result try_swap(float t,
             /* Update clb data structures since we kept the move. */
             commit_move_blocks(blocks_affected);
 
-        } else { /* Move was rejected.  */
-                 /* Reset the net cost function flags first. */
+        } else { //move_outcome == REJECTED
+
+            /* Reset the net cost function flags first. */
             reset_move_nets(num_nets_affected);
 
             /* Restore the place_ctx.block_locs data structures to their state before the move. */
             revert_move_blocks(blocks_affected);
 
+            if (place_algorithm == SETUP_SLACK_ANALYSIS_PLACE) {
+                //Revert the timing delays and costs to pre-update values
+                //These routines must be called after reverting the block moves
+                //TODO: make this process incremental
```

Comment:
Is this TODO still relevant or are they incremental already?  They look incremental (with the invalidator etc.).